src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -21,9 +25,29 @@ public DeleteCommand parse(String args) throws ParseException {
             Index index = ParserUtil.parseIndex(args);
             return new DeleteCommand(index);
         } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), pe);
+            ArgumentMultimap argMultimap = checkValidity(args);
+
+            try {
+                String str = argMultimap.getValue(PREFIX_NAME).orElseThrow(() -> new Exception("shouldnt be thrown"));
+                Name name = ParserUtil.parseName(str);
+                return new DeleteCommand(name);
+            } catch (Exception ex) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), ex);
+            }
         }
```

Comment:
I think you can use `ArgumentMultimap argMultimap =
                ArgumentTokenizer.tokenize(args, PREFIX_NAME)` to check if users want to delete by name or delete by index **but not both** so that you can avoid this code? However, if there is really a situation which you cannot avoid this, maybe you can change something like this, would this be more readable?
```java
...
} catch (ParseException unusedException) {}
ArgumentMultimap argMultimap = checkValidity(args);

try {
   String str = argMultimap.getValue(PREFIX_NAME);
   Name name = ParserUtil.parseName(str);
   return new DeleteCommand(name);
} catch (Exception ex) {
   throw new ParseException(
                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), ex);
}
...
```
It basically extracts everything from the first `catch (ParseException pe)` and leaves it empty so that you can reduce one nested layer.

Reply:
I initially thought about having empty catch blocks -- but checkstyle will fail it

Hence I tweaked it actually.